openedx/features/content_tagging/tests/test_tasks.py

```diff
@@ -0,0 +1,117 @@
+"""
+Test for auto-tagging content
+"""
+from unittest.mock import patch
+
+import ddt
+from django.conf import settings
+from django.core.management import call_command
+from django.test.utils import override_settings
+from openedx_tagging.core.tagging.models import ObjectTag, Taxonomy
+from organizations.models import Organization
+
+from openedx.core.lib.tests import attr
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.tests.test_mixed_modulestore import CommonMixedModuleStoreSetup
+
+from ..models import ContentLanguageTaxonomy, TaxonomyOrg
+from ..tasks import delete_xblock_tags, update_course_tags, update_xblock_tags
+
+if not settings.configured:
+    settings.configure()
+
+LANGUAGE_TAXONOMY_ID = -1
+
+
+@ddt.ddt
+@attr("mongo")
+@override_settings(
+    CELERY_EAGER_PROPAGATES_EXCEPTIONS=True,
+    CELERY_ALWAYS_EAGER=True,
+    BROKER_BACKEND="memory",
+)
+class TestCourseAutoTagging(CommonMixedModuleStoreSetup):
+    """
+    Test if the handlers are callend and if they call the right tasks
+    """
+
+    def _check_tag(self, object_id, taxonomy, value):
+        object_tag = ObjectTag.objects.filter(object_id=object_id, taxonomy=taxonomy).first()
+        assert object_tag, "Tag not found"
+        assert object_tag.value == value, f"Tag value mismatch {object_tag.value} != {value}"
+        return True
+
+    @classmethod
+    def setUpClass(cls):
+        # Run fixtures to create the system defined tags
+        super().setUpClass()
+        call_command("loaddata", "--app=oel_tagging", "language_taxonomy.yaml")
+
+        # Configure language taxonomy
+        language_taxonomy = Taxonomy.objects.get(id=-1)
+        language_taxonomy.taxonomy_class = ContentLanguageTaxonomy
+        language_taxonomy.save()
+        TaxonomyOrg.objects.create(id=-1, taxonomy_id=-1, org=None)
+
+    def setUp(self):
+        super().setUp()
+        self.orgA = Organization.objects.create(name="Organization A", short_name="orgA")
+
+    @ddt.data(
+        ModuleStoreEnum.Type.mongo,
+        ModuleStoreEnum.Type.split,
+    )
+    @patch("openedx.features.content_tagging.tasks.modulestore")
+    def test_create_course_with_xblock(self, default_ms, mock_modulestore):
```

Comment:
This test has a lot of mocking and patching so it's hard to tell what's actually working. Since [celery is eager by default in tests](https://github.com/openedx/edx-platform/blob/a013c08ae6beea4e3e8ef91cb12ffe32af5b9b33/cms/envs/test.py#L202), can't you simplify this test a lot to remove all the mocks and patches, and just : create a course, create an XBlock, check that the tags get created, etc.

Reply:
And I think it would be better to separate the tests for the courses and the tests for the xblocks so that it is easy to see what is being tested.

Reply:
Sure! Will do!

Reply:
Done!